idp/static/static.go

```diff
@@ -194,9 +197,16 @@ func (idp *identityProvider) Handle(ctx context.Context, w http.ResponseWriter,
 		id, err := idputil.HandleLoginForm(ctx, w, req, idpChoice, idp.initParams.Template, idp.loginUser)
 		if err != nil {
 			idp.initParams.VisitCompleter.RedirectFailure(ctx, w, req, ls.ReturnTo, ls.State, err)
+			return
 		}
-		if id != nil {
-			idp.initParams.VisitCompleter.RedirectSuccess(ctx, w, req, ls.ReturnTo, ls.State, id)
+		if id != nil && idp.params.Require2FA {
+			mfaState, err := idp.initParams.SetMFAStateProviderID(w, string(id.ProviderID))
```

Comment:
could this part be done in RedirectMFA? to save every IDP from having to do the same thing?

Reply:
done